unified-prototype/unified-plugin/plugin-android/src/main/java/org/gradle/api/experimental/android/AndroidDSLSupport.java

```diff
@@ -16,109 +16,15 @@
 
 package org.gradle.api.experimental.android;
 
-import com.android.build.api.dsl.LibraryExtension;
-import com.android.build.api.variant.AndroidComponentsExtension;
 import org.gradle.api.Action;
-import org.gradle.api.JavaVersion;
-import org.gradle.api.Project;
 import org.gradle.api.provider.Property;
-import org.jetbrains.kotlin.gradle.dsl.KotlinAndroidProjectExtension;
 
-import javax.annotation.Nullable;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
-import java.util.Set;
-
-public class AndroidDSLSupport {
```

Comment:
❓ Why remove the common utilities into 2 separate classes?  This code is identical or very similar between libraries and applications.

Reply:
1. They were not really common at all. The methods were only called in one place so I didn't see a reason to pull them out to another class. 
2. While both plugins use very similar logic, they cannot share logic due to the separate type hierarchies. If we end up merging the type hierarchies in the future I can see value in pulling out the logic to a common location but as it stands now each separate implementation of these linking methods are only called from their respective plugin.